pkg/manifests/config.go

```diff
@@ -35,14 +35,18 @@ type Config struct {
 }
 
 type PrometheusOperatorConfig struct {
-	BaseImage                string `json:"baseImage"`
-	PrometheusConfigReloader string `json:"prometheusConfigReloaderBaseImage"`
-	ConfigReloaderImage      string `json:"configReloaderBaseImage"`
+	BaseImage                   string `json:"baseImage"`
+	Tag                         string `json:"-"`
```

Comment:
When is this struct ever marshalled to json? Why exclude the `Tag`?

Reply:
It's actually for the other way around, so that when this struct is unmarshaled, users can't use the config struct to set the tag. Instead the tag is enforced by the flag. Basically this means that the flag is configured for the lifetime of the version whereas the config can be adapted at runtime.

Reply:
:+1: neat!